src/io/flutter/sdk/FlutterSdkUtil.java

```diff
@@ -186,7 +186,6 @@ public static String getErrorMessageIfWrongSdkRootPath(final @NotNull String sdk
 
   public static boolean isFlutterProjectDir(@Nullable VirtualFile dir) {
     if (dir == null || !dir.isDirectory()) return false;
-    if (dir.findChild(FlutterConstants.FLUTTER_YAML) != null) return true;
```

Comment:
And here too.  Net-net: I think the check was always fishy once we started reading pubspecs for Flutter dependencies.  Anyway, see my hypothesis below.